src/components/select-field/BaseSelectField.js

```diff
@@ -85,6 +85,24 @@ class BaseSelectField extends React.Component<Props, State> {
             isOpen: false,
             shouldScrollIntoView: false,
         };
+
+        this.fieldRef = React.createRef();
+    }
+
+    componentDidMount() {
+        this.calculateOffsetTopValue();
+    }
+
+    /**
+     * Calculate this field's distance from the top.
+     * This allows us to create an overflow container for the field's dropdown menu that is navigable using both
+     * the mouse and arrow keys and will always fit within the current viewport.
+     */
+    calculateOffsetTopValue() {
+        this.offsetTopValue =
+            window.pageYOffset +
+            this.fieldRef.current.getBoundingClientRect().top +
+            BASE_SELECT_FIELD_OFFSET_TOP_PADDING;
```

Comment:
Can we add this via CSS as padding or margin rather than including it in the height calculation?